### PR TITLE
Address issues raised by recent reviews

### PR DESCRIPTION
**Pull request by aphillips**
_editor's copy locations include the fragment reference and are linked to @aphillips's repo version, since that displays more nicely than the diff/preview)_

Addresses #88
Add guidance for generating bidirectional labels/descriptions
Caused by w3c/i18n-activity#1638
[editor's copy location #strcat](https://aphillips.github.io/bp-i18n-specdev/#strcat)

Addresses #83
Add guidance on Unicode code point order
[editor's copy location #char_sort](https://aphillips.github.io/bp-i18n-specdev/#char_sort)

Addresses #69
Add information about constituent parts of BCP47 (specifically 4647)
[editor's copy location #lang_bcp_not_rfc](https://aphillips.github.io/bp-i18n-specdev/#lang_bcp_not_rfc)
[editor's copy location #lang_matching_bcp](https://aphillips.github.io/bp-i18n-specdev/#lang_matching_bcp)

Addresses #14 
Add guidance on using the ILSTR instead of making your own list of subtags
[editor's copy location #use_lstr](https://aphillips.github.io/bp-i18n-specdev/#use_lstr)

***
<a href="https://pr-preview.s3.amazonaws.com/aphillips/bp-i18n-specdev/pull/89.html" title="Last updated on Mar 16, 2023, 5:23 PM UTC (6ce0aea)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/bp-i18n-specdev/89/3e53a99...aphillips:6ce0aea.html" title="Last updated on Mar 16, 2023, 5:23 PM UTC (6ce0aea)">Diff</a>

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -351,7 +351,15 @@ <h3>Defining language values</h3>
...
+ 	
+ 	<aside class="issue"><p>The topic of matching language tags needs more detail, and may merit being a separate section.</p></aside>
+ 	
+ 	<div class="req" id="use_lstr">
+		<p class="advisement">Specifications SHOULD refer to the IANA Language Subtag Registry instead of providing lists of codes extracted from ISO 639, ISO 3166, or other standards.</p>
+ 	</div>
+ 	
+ 	<p>As part of BCP 47, IANA maintains the language subtag registry, which is a publicly available, machine-readable list of valid subtags for use in language tags. While this registry is based on underlying ISO standards, such as ISO 639 (languages) and ISO 3166 (regions), the list is actively maintained, stabilized, and comprehensive in ways that other lists found on the Internet may not be. Each of the subtag types is kept in sync with parent standards with the help and participation of those standards maintainers. These include the various parts of ISO 639 (639-1, 639-2, 639-3), the ISO 15924 script codes, and ISO 3166 and UN M.49 region codes. Extracting or making your own list of codes or referring to ones found elsewhere can lead to maintenance problems or confusion.</p>
```
Comment (r12a) [inline index.html]: You could also add something like:
"Selecting one of the ISO or UN code lists will unnecessarily restrict the list of languages that can be described."

File: index.html [hunk 2]
```diff
@@ -951,15 +959,15 @@ <h3>Generating or requiring creation of mixed direction strings</h3>
...
+<p>Notice how the part number (<kbd lang="zxx" translate="no">A123B</kbd>) is separated from the brand name (<kbd lang="ar" dir="rtl" translate="no">&#x0645;&#x0627;&#x0631;&#x0643;&#x0629;</kbd>), the measurement <kbd>36</kbd> and the marker for inches (<kbd>"</kbd>) have become separated and that the values <kbd>75 Hz</kbd> (where the measurement is in ASCII) and <kbd>4 ms</kbd> (where the measurement has an Arabic translation <kbd lang="ar" dir="rtl" translate="no">&#x0645;&#x0644;&#x0644;&#x064A; &#x062B;&#x0627;&#x0646;&#x064A;&#x0629;</kbd>) both separate the number from the measurement. Generating labels from a sequence of string tokens requires extra care to ensure that the complete string is "bidirectionally clean" and will display properly to the user. Adding isolating bidirectional controls to the above string produces better results:</p>
```
Comment (r12a) [inline index.html]: If "isolating bidirectional controls" means either markup or formatting characters, this should probably be mentioned.

**Review by r12a — CHANGES_REQUESTED** [review]

File: index.html (hunk 1 above)
Comment (aphillips) [reply index.html]: Good callout. I did a more extensive edit, plus I added links to the "choosing a language tag" article, your subtag lookup tool, and the registry itself.

File: index.html (hunk 2 above)
Comment (aphillips) [reply index.html]: Done.

**Comment by xfq** [comment]
I noticed that Netlify wasn't working. It seems like someone revoked access to all W3C repos (not just the i18n ones). I'll try to add it back.

File: local.css [hunk 3]
```diff
@@ -431,3 +431,9 @@ table.whitespace caption {
   background-color: #04AA6D;
   color: white;
 }
+
+td.exampleChar {
+  font-size: 36pt;
+  text-align: center;
+  vertical-align: center;
```
Comment (xfq) [inline local.css]: This should probably be `vertical-align: middle;`. See https://w3c.github.io/csswg-drafts/css2/#propdef-vertical-align

Comment (aphillips) [reply local.css]: thanks. Fixed in 07cbc60.

**Comment by aphillips** [comment]
@r12a Could you look at this for merging-readiness? I'd like to include these changes when pushing to TR...

File: index.html [hunk 4]
```diff
@@ -937,9 +947,58 @@ <h3><em>Detecting &amp; matching direction (TBD)</em></h3>
...
 <p class="reviewComments"><a href="https://github.com/w3c/i18n-activity/labels/t%3Abidi_detection_x" target="_blank">See related review comments.</a></p>
 </section>
-</section>
 
+<section id="strcat" class="subtopic">
+<h3>Concatenation of strings</h3>
 
+<p class="reviewComments"><a href="https://github.com/w3c/i18n-activity/labels/t%3Astrcat" target="_blank">See related review comments.</a></p>
+
+<div class="req" id="strcat_anti_pattern">
```
Comment (r12a) [inline index.html:956]: I don't think people will find this general advice about composite messages, that has nothing to do with text direction.

I think it is best to move the section "Composite Strings" under 'Text-processing" and then add a 'see also' to the text direction section 3.6.

File: index.html [hunk 5]
```diff
@@ -937,9 +947,58 @@ <h3><em>Detecting &amp; matching direction (TBD)</em></h3>
...
+</div>
+
+<p>Creation of [=natural language=] text values by concatenating multiple strings together is an internationalization anti-pattern. Languages vary greatly in word order, count, grammatical gender or case, punctuation, and many other requirements. As a result, avoid requiring or suggesting that implementations generate human-readable messages from sub-strings.</p>
+
+<div class="xref"><span class="seealso">See also</span>
+   <p><cite><a href="https://www.w3.org/International/articles/composite-messages/index.en.html">Working with Composite Messages</a></cite></p>
+</div>
+
+<div class="req" id="bidi_gen_advice">
+	<p class="advisement">Specifications SHOULD provide guidance to implementers on avoiding problems with text directionality for any field(s) that the specification requires an implementation to create or generate for display to users.</p>
```
Comment (r12a) [inline index.html:967]: Rather a complicated and long sentence.

How about:
When a specification requires an implementation to create or generate text which will be displayed to users, the specification should provide implementers with guidance on
how to avoid potential problems related to text direction.

File: index.html [hunk 6]
```diff
@@ -937,9 +947,58 @@ <h3><em>Detecting &amp; matching direction (TBD)</em></h3>
...
+</div>
+
+<p>Specifications for APIs, protocols, or document formats sometimes require an implementation to create or provide a field containing a display name or description. When such a string is assembled from separate parts, it can result in problems with presentation or understanding due to the way that the <cite>Unicode Bidirectional Algorithm</cite> [[UAX9]] processes the assembled string. In such cases, the specification should guide implementers about how to create values that will display properly.</p>
```
Comment (r12a) [inline index.html:970]: Several things are unclear to me here, and rather than provide text suggestions that may be barking up the wrong tree, we should perhaps discuss during our telecon.  Questions include: 
- do you mean bidrectional text or text that generally runs in a particular direction (eg. RTL)?
- i suspect, but i'm not sure, that we are talking solely about strings that contain bidirectional text within them, or about correctly displaying any string that is inserted into a target location by the consumer?
- if the latter, we're focusing on what producers need to do, we should mention dirname or using other metadata

Basically, i think we need a little more clarity about the role of producers vs consumers, and in-string direction vs how a string will be inserted into a target location.  It seems a little vague at the moment.  The section title, Concatenation of Strings, doesn't make it sound like we're concerned with string internal direction, but rather strings being inserted into the target location.  The example shows one scenario that fits the section title, but the preceding text needs to be clearer before we get to the example, and needs to clarify whether we are also talking about string-internals.

File: index.html [hunk 7]
```diff
@@ -2779,42 +2838,103 @@ <h3>Specifying sort and search functionality</h3>
...
+	<li><p class="link"><a href="https://unicode.org/reports/tr10">Unicode Collation Algorithm</a> [[UTS10]]</p></li>
+    <li class="w3"><p class="link"><a href="https://www.w3.org/TR/charmod/#sec-CollationUnits">Units of collation</a>, in <cite>Character Model for the World Wide Web: Fundamentals</cite>.</p></li>
 </ul>
 </aside>
 
-
-    <p>Applications often need to organize sets of information or content. Frequently this involves sorting the content so that users can find what they are looking for. Many data types, such as numbers or dates, are easily sorted by comparing the values. When it comes to textual information, however, the nature of character encodings brings some additional complexity.</p>
+    <p>Applications often need to organize sets of information or content. Frequently this involves sorting the content. Many data types, such as numbers or dates, are easily sorted by comparing the values. When it comes to textual information, however, the nature of character encodings and user expectations regarding "alphabetical" order brings some additional complexity.</p>
```
Comment (r12a) [inline index.html:2847]: Numbers and dates are only easy to sort if you compare internal representations of the data.

File: index.html [hunk 8]
```diff
@@ -2828,6 +2948,12 @@ <h3>Specifying sort and search functionality</h3>
 	<p><a href="https://www.w3.org/TR/charmod/#sec-CollationUnits">Units of collation, C008</a>, in <cite>Character Model for the World Wide Web: Fundamentals</cite></p>
 	</details>
 	</div>
+-->
+
+	<aside class="warning" id="find-is-hard" title="String search is hard">
```
Comment (r12a) [inline index.html:2953]: As a warning about _string searching_ this seems a little oddly placed here.  I think that similar difficulties apply to matching strings for sorting, but i think it would be useful to couch this warning in those terms.

File: local.css [hunk 9]
```diff
@@ -431,3 +431,9 @@ table.whitespace caption {
   background-color: #04AA6D;
   color: white;
 }
+
+td.exampleChar {
+  font-size: 36pt;
```
Comment (r12a) [inline local.css:436]: Can we use a rem or, if necessary, an em unit rather than pt?  If you really want to fix the size, perhaps use px rather than pt.

File: index.html (hunk 7 above)
Comment (aphillips) [reply index.html:2847]: Done with:

```suggestion
Many non-textual data types, such as numbers or dates, can be easily sorted using the internal data representation. 
```

File: index.html (hunk 8 above)
Comment (aphillips) [reply index.html:2953]: This section is (and has historically been) called "specifying sort and search functionality". All of the prose in the section is about _sorting_. This is the only mention of "search"--I added it so that we wouldn't have a gap. I think maybe a subsection div above it would help. I also retitled as "under construction". See the new PR

File: local.css (hunk 9 above)
Comment (aphillips) [reply local.css:436]: Done. Made it `300%`

File: index.html (hunk 5 above)
Comment (aphillips) [reply index.html:967]: Done.

File: index.html (hunk 6 above)
Comment (aphillips) [reply index.html:970]: added to the agenda for 2023-03-30

File: index.html (hunk 4 above)
Comment (aphillips) [reply index.html:956]: pending telecon discuss
